FirebaseAuth/Sources/SystemService/FIRSecureTokenService.m

```diff
@@ -170,6 +172,20 @@ - (void)requestAccessToken:(FIRFetchAccessTokenCallback)callback {
            BOOL tokenUpdated = NO;
            NSString *newAccessToken = response.accessToken;
            if (newAccessToken.length && ![newAccessToken isEqualToString:self->_accessToken]) {
+             FIRAuthTokenResult *tokenResult =
+                 [FIRAuthTokenResult tokenResultWithToken:newAccessToken];
+             // There is an edge case where the request for a new access token may be made right
+             // before the app goes inactive, resulting in the callback being invoked much later
+             // with an expired access token. This does not fully solve the issue, as if the
+             // callback is invoked less than an hour after the request is made, a token is not
+             // re-requested here but the approximateExpirationDate will still be off since that is
+             // computed at the time the token is received.
+             if (retryIfExpired && [tokenResult.expirationDate timeIntervalSinceNow] <= 0) {
```

Comment:
Would it be worth re-fetching the token if the token will expire within 5 minutes as well? (Instead of only after it's expired)

Reply:
Good call, done.